numpy/lib/function_base.py

```diff
@@ -1624,46 +1624,45 @@ def trim_zeros(
     """
     trim = trim.lower()
 
+    absolutes = np.abs(filt)
+    lengths = np.zeros((absolutes.ndim, 2), dtype=np.intp)
+
     if axis is None:
         # Apply iteratively to all axes
-        axis = range(filt.ndim)
-
+        axis = range(absolutes.ndim)
     # Normalize axes to 1D-array
     axis = np.asarray(axis, dtype=np.intp)
     if axis.ndim == 0:
         axis = np.asarray([axis], dtype=np.intp)
 
-    absolutes = np.abs(filt)
-    lengths = np.zeros((absolutes.ndim, 2), dtype=np.intp)
+    if atol > 0:
+        absolutes[absolutes <= atol] = 0
+    if rtol > 0:
+        absolutes[absolutes <= rtol * absolutes.max()] = 0
+
+    nonzero = np.nonzero(absolutes)
 
     for current_axis in axis:
         absolutes.take([], current_axis)  # Raises if axis is out of bounds
```

Comment:
Better to just call `axis = normalize_axis_index(axis, absolute.ndim)` here

Reply:
Nice, that function is indeed exactly what I needed.